articles/language/file-structure.md

```diff
@@ -385,3 +385,32 @@ function DotProduct(a : Double[], b : Double[]) : Double {
     return accum;
 }
 ```
+
+
+## Internal Declarations
+
+User-defined types, operations, and functions can also be declared as *internal*.
+This means that they can only be accessed from within the Q# project that they are declared in.
+When a project is used as a reference, all of its *public* (non-internal) declarations are made available, but trying to use an internal declaration from another project will produce an error.
+Internal declarations are useful for writing modular code that can be reused by other parts of your project, but still be changed later without breaking other projects that might depend on it.
+
+> [!WARNING]
+> Internal user-defined types cannot be used in the signatures of public callables or as underlying types of public user-defined types.
+> They can only be used in signatures or underlying types if the corresponding callable or user-defined type is also internal.
+
+An internal user-defined type, operation, or function can be declared simply by adding `internal` at the beginning of the declaration.
+For example,
+
+```qsharp
+internal newtype PairOfInts = (Int, Int);
```

Comment:
It would be great if the syntax highlighting showed `internal` in the same color as `newtype`, `operation`, etc., but I'm not sure how to change that.

Reply:
@geduardo @gillenhaalb @Bradben any suggestion how to do this?

Reply:
@cgranade Do you know where the language json for the docs page is?

Reply:
I'm not quite sure, sorry. It looks like highlight.js is being used server-side to generate syntax highlighting, but Q# isn't a supported language for highlight.js, so I'm not sure where the right plugin is being loaded. @natke or @Bradben may know more.

Reply:
Also pinging @tcNickolas; any ideas?

Reply:
No, sorry, we set it up about two years ago and it looks like I wasn't involved. Somebody from the docs team should know, I remember we relied on them for setting this up.

Reply:
@Bradben @natke The colorization for the newly introduced keyword "internal" is not currently matching the one for the existing keywords "operation", "function", and "newtype". the current syntax highlighting reflects Q# syntax as of November 2017 (Q# 0.1), and not any recent changes; keywords like “Unit,” “is,” “Adj,” etc. are also not highlighted.

Reply:
Possibly we could also create a separate issue for the syntax highlighting so this PR isn't blocked by it.

Reply:
I agree, @marshallsa. Creating now.

Reply:
Issue #762 - thanks @cgranade! I'm going to resolve this thread then.

Reply:
No worries, happy to help! I just used the handy "Reference as new issue" feature, really...

![image](https://user-images.githubusercontent.com/31516/77356084-e5252380-6d02-11ea-8bd7-6c9e7d6f4267.png)

Reply:
Cool, I forgot about that feature. :)